frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx

```diff
@@ -0,0 +1,102 @@
+import {
+  humanizePercentage,
+  humanizeBinaryBytes,
+  humanizeNumber,
+} from '@console/internal/components/utils';
+
+import { BlockPoolDashboardContext } from './block-pool-dashboard-context';
+import { EfficiencyItemBody } from '../common/storage-efficiency/storage-efficiency-card-item';
+
+import { getPoolQuery, StorageDashboardQuery } from '../../../queries/ceph-queries';
+
+const parser = compose((val) => val?.[0]?.y, getInstantVectorStats);
+
+export const CompressionDetailsCard: React.FC = () => {
+  const { t } = useTranslation();
+  const { obj } = React.useContext(BlockPoolDashboardContext);
+  const compressionMode = obj.spec?.compressionMode;
+  const compressionEnabled = compressionMode !== 'none';
+  const { name } = obj.metadata;
+
+  // Compression Metrics
+  const queries = React.useMemo(
+    () => [
+      getPoolQuery([name], StorageDashboardQuery.POOL_COMPRESSION_SAVINGS),
+      getPoolQuery([name], StorageDashboardQuery.POOL_COMPRESSION_ELIGIBILITY),
+      getPoolQuery([name], StorageDashboardQuery.POOL_COMPRESSION_RATIO),
+    ],
+    [name],
+  );
+
+  const [values, loading, loadError] = usePrometheusQueries(queries, parser as any);
+  const poolCompressionSavings = values?.[0];
+  const poolCompressionEligibility = values?.[1];
+  const poolCompressionRatio = values?.[2];
+
+  const compressionEligibilityProps = {
+    stats: Number(poolCompressionEligibility),
+    isLoading: loading,
+    error: loadError || !poolCompressionEligibility,
+    title: t('ceph-storage-plugin~Compression eligibility'),
+    getStats: () => humanizePercentage(poolCompressionEligibility).string,
+    infoText: 'indicates the percentage of incoming data that is compressible',
+  };
+
+  const compressionSavingsProps = {
+    stats: Number(poolCompressionSavings),
+    isLoading: loading,
+    error: loadError || !poolCompressionSavings,
+    title: t('ceph-storage-plugin~Compression savings'),
+    getStats: () => humanizeBinaryBytes(poolCompressionSavings).string,
+    infoText:
+      'indicates the total savings gained from compression for this pool, including replicas',
+  };
+
+  const compressionRatioProps = {
+    stats: Number(poolCompressionRatio),
+    isLoading: loading,
+    error: loadError || !poolCompressionRatio,
+    title: t('ceph-storage-plugin~Compression ratio'),
+    getStats: () => `${humanizeNumber(poolCompressionRatio).string}:1`,
+    infoText: 'indicates the achieved compression on eligible data for this pool',
+  };
+
+  return (
+    <DashboardCard data-test-id="compression-details-card">
+      <DashboardCardHeader>
+        <DashboardCardTitle>{t('ceph-storage-plugin~Compression')}</DashboardCardTitle>
+      </DashboardCardHeader>
+      <DashboardCardBody>
+        <DetailsBody>
+          <DetailItem isLoading={!obj} title={t('ceph-storage-plugin~Compression status')}>
+            {!compressionEnabled ? 'Disabled' : 'Enabled'}
+          </DetailItem>
+        </DetailsBody>
+        <DetailsBody>
```

Comment:
remove \<DetailsBody\> from here

Reply:
Made some changes according to @vbnrh comments. Please take a look (and please see the screenshots provided below as well)

Reply:
```suggestion
```